### ISSUE TITLE
Exceptions inside CommandOutcomes are not reported by Bootique

### ISSUE THREAD

**Issue by andrus** [bootique/bootique#148]
This is a 0.23-only problem that is a result of #25. Currently `Bootoque.exec(..) `stopped printing stack traces of exceptions coming inside CommandOutcome objects from commands (such as Liquibase UpdateCommand). I guess we do need to report such exceptions via BootLogger.stderr. 

Also this raises a question of null or incomplete "message" properties in CommandOutcome produced by  Liquibase UpdateCommand (and maybe other commands).

**Comment by andrus** [comment]
Another example from bootique-jersey. I created an incorrect configuration:

```yaml
jersey:
  # missing leading slash in urlPattern
  urlPattern: api
```

The app died with no indication what so ever as to what went wrong:

```
INFO i.b.j.s.ServerFactory: Adding listener io.bootique.jetty.servlet.DefaultServletEnvironment
INFO i.b.j.s.ServletFactory: Adding servlet 'jersey' mapped to api
INFO i.b.j.s.FilterFactory: Adding filter 'bootiqueShiro' mapped to /*
INFO i.b.j.s.ServerLifecycleLogger: Starting jetty...
INFO o.e.j.server.Server: jetty-9.3.6.v20151106
INFO o.e.j.s.ServerConnector: Started ServerConnector@44f9779c{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}
Error running command '-s -c app.yml'
```

**Comment by andrus** [comment]
Another subtle improvement - when an exception or its cause is a BootiqueException, we should avoid logging stack trace (BootiqueException's message and error code are important; its location is irrelevant). Fixed per 1cf36b0.
